build.gradle.kts

```
@@ -0,0 +1,232 @@
+    testSourceSets(sourceSets["functionalTest"], sourceSets["crossScalaVersionTest"])
+}
+
+val kotlindocJar by tasks.registering(Jar::class) {
+    from(tasks.dokkaHtml.get().outputDirectory)
+    archiveClassifier.set("kotlindoc")
+    dependsOn(tasks.dokkaHtml)
+}
+
+val sourcesJar by tasks.registering(Jar::class) {
+    from(sourceSets["main"].allSource)
+    archiveClassifier.set("sources")
+}
+
+fun propOrDefault(property: String): String {
+    if (project.hasProperty(property)) {
+        return project.property(property).toString()
+    } else {
+        return ""
+    }
+}
+
+configure<PublishingExtension> {
+    repositories {
+        maven {
+            url = uri("https://oss.sonatype.org/service/local/staging/deploy/maven2/")
+            credentials {
+                username = propOrDefault("sonatypeUser")
+                password = propOrDefault("sonatypePass")
+            }
+        }
+    }
+    publications {
+        create<MavenPublication>("mavenJava") {
+            pom {
+                name.set("GradleScoverage")
+                description.set(description)
+                url.set(website)
+
+                scm {
+                    url.set(scmUrl)
+                    developerConnection.set(scmUrl)
+                }
+
+                licenses {
+                    license {
+                        name.set("The Apache Software License, Version 2.0")
+                        url.set("http://www.apache.org/licenses/LICENSE-2.0.txt")
+                        distribution.set("repo")
+                    }
+                }
+
+                developers {
+                    developer {
+                        id.set("maiflai")
+                    }
+                    developer {
+                        id.set("ubourdon")
+                    }
+                    developer {
+                        id.set("D-Roch")
+                    }
+                    developer {
+                        id.set("eyalroth")
+                    }
+                }
+            }
+            from(components["java"])
+            artifact(kotlindocJar)
+            artifact(sourcesJar)
+        }
+    }
+}
+
+if (project.properties.containsKey("signing.keyId")) {
+    apply(plugin = "signing")
+    configure<SigningExtension> {
+        sign(the<PublishingExtension>().publications["mavenJava"])
+    }
+}
+
+// see https://stackoverflow.com/questions/44679007
+val fixIdeaPluginClasspath by tasks.registering {
+    doFirst {
+        tasks {
+            named<PluginUnderTestMetadata>("pluginUnderTestMetadata") {
+                val ideaClassesPath = project.buildDir.toPath().resolveSibling("out").resolve("production")
+                val newClasspath = pluginClasspath.toMutableList()
+                newClasspath.add(0, file(ideaClassesPath))
+                pluginClasspath.setFrom(newClasspath)
+            }
+        }
+    }
+}
+tasks["pluginUnderTestMetadata"].mustRunAfter(fixIdeaPluginClasspath)
```

```suggestion
tasks.pluginUnderTestMetadata {
    mustRunAfter(fixIdeaPluginClasspath)
}
```
That being said, this mechanism seems very strange - I developed a lot of Gradle plugins using IDEA, and never had to modify idea settings in this way. Are you sure it doesn't work without this?

---

I read why this happens on the attached stackoverflow link, and I think I see the issue (running integration tests from IDEA), but the more idiomatic solution would be to run tests via Gradle instead of via IDEA (it is mentioned in one of the answers in that stackoveflow questsion). Just something to consider :)